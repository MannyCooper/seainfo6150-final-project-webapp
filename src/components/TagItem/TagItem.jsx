import React from "react";
import { Link } from "react-router-dom";
import styles from "./TagItem.module.css"

const TagItem = (props) =>{
    let path = '/tags/' + props.tag[0];
    return (
        <div className={styles.tag}>
            <Link className={styles.link} to={path} >
                <span className={styles.name}>{props.tag[0]}</span>
                <span className={styles.number}>{props.tag[1]}</span>
            </Link>
        </div>
    )
}

export default TagItem;