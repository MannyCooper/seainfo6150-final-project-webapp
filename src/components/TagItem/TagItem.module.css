.tag{
    margin-right: 20px;
    transition: transform .4s cubic-bezier(.4,0,.25,1);
}

.tag:hover{
    -webkit-transform: translateY(-6px);
    transform: translateY(-6px);
    -webkit-transform: scale(1.3);
    transform: scale(1.1);
}

.name{
    background: #f2f2f7;
    color: #3a3a3c;
    justify-content: center;
    line-height: 4;
    white-space: nowrap;
    border-radius: 15px 0 0 15px;
    padding: 10px 15px
}

.number{
    background: #e5e5ea;
    color: #48484a;
    border-radius: 0 15px 15px 0;
    padding: 10px 15px
}

@media only screen and (max-width: 1200px){
    .name,.number{
        line-height: 3;
        padding: 5px 10px
    }
}

@media only screen and (max-width: 768px){
    .name,.number{
        line-height: 2.5;
        padding: 4px 8px
    }
}   
