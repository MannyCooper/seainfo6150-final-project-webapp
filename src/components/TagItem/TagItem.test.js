 /* global render */
 import React from "react";
 import TagItem from "./TagItem.jsx";
 
 const testTag = ['Test',10]
 
 it("renders the TagItem component correctly", () => {
   const { container } = render(<TagItem tag={testTag} />);
   expect(container).toMatchSnapshot();
 });
 