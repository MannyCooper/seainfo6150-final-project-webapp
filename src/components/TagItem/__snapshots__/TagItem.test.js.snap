// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TagItem component correctly 1`] = `
<div>
  <div
    class="tag"
  >
    <a
      class="link"
      href="/tags/Test"
    >
      <span
        class="name"
      >
        Test
      </span>
      <span
        class="number"
      >
        10
      </span>
    </a>
  </div>
</div>
`;
